Add PDS Base Image to FindSecurityBugs Docker Compose files

ddzikon opened the pull request:
This PR adds `${BASE_IMAGE}` to following Docker Compose files for FindSecurityBugs:  
* docker-compose_pds_findsecuritybugs.yaml
* docker-compose_pds_findsecuritybugs_cluster.yaml
* docker-compose_pds_findsecuritybugs_cluster_object_storage.yaml
* docker-compose_pds_findsecuritybugs_external-network.yaml

Value for `BASE_IMAGE` can be set in files containing initial values:
* `env-initial`
* `env-cluster-initial`

closes #1860

ddzikon commented:
There is a slight difference between versions:
* `PDS_VERSION=0.31.0`
* `BASE_IMAGE="ghcr.io/mercedes-benz/sechub/pds-base:v0.32.1"`

I see that most images using `pds-base` consider version `0.32.1`.

Is it ok to have it so or should it be addressed?

Jeeppler commented:
@ddzikon thanks for your contribution.

The local docker-compose files for the FindSecurityBugs solution are indeed not working.

We introduced the PDS base image, after we figured out that we can: a) push image to ghcr.io and that b) images can be hosted publicly on ghcr.io (ghcr.io is the second attempt of GitHub to create a container registry).

Variables like the `PDS_VERSION` are already set in the PDS-Base image and are therefore redundant. All variables set in the base image are inherited by all PDS-Solutions using the base image. So, please have a look at the Dockerfile of the PDS-Base image: https://github.com/mercedes-benz/sechub/blob/develop/sechub-pds-solutions/pds-base/docker/PDS-Debian.dockerfile#L122. That means the `PDS_VERSION` variable can be removed, it is enough to use the `BASE_IMAGE` going forward.

Another simplification which we did some time ago, was to simply reference the standard environment files. As they are all the same. Have a look at the file from GoSec: https://github.com/mercedes-benz/sechub/blob/develop/sechub-pds-solutions/gosec/01-start-single-docker-compose.sh:

~~~
ENVIRONMENT_FILES_FOLDER="../shared/environment"
ENVIRONMENT_FILE=".env-single"

# Only variables from .env can be used in the Docker-Compose file
# all other variables are only available in the container
setup_environment_file ".env" "env"
setup_environment_file "$ENVIRONMENT_FILE" "$ENVIRONMENT_FILES_FOLDER/env-base"
~~~

Like in GoSec, there should only be a single "env" file with the base image variable in it. The 01… scripts copy the values from the shared files and create an `.env` and a `.env-single` etc. file.

Please incorporate those changes into your pull-request.

Jeeppler commented on sechub-pds-solutions/findsecuritybugs/docker-compose_pds_findsecuritybugs.yaml:
```diff
@@ -5,7 +5,7 @@ services:
             args:
-                - BASE_IMAGE=debian:11-slim
+                - BASE_IMAGE=${BASE_IMAGE}
                 - PDS_VERSION=${PDS_VERSION}
```
Please remove the `PDS_VERSION` variable: [#1862#issuecomment-1382224019](https://github.com/mercedes-benz/sechub/pull/1862#issuecomment-1382224019)

Jeeppler commented on sechub-pds-solutions/findsecuritybugs/env-cluster-initial:
```diff
@@ -19,4 +19,5 @@ S3_ACCESSKEY=user
 S3_SECRETKEY=Why$5forThat?
```
This file should be removed. For more information have a look at: https://github.com/mercedes-benz/sechub/pull/1862#issuecomment-1382224019.

Jeeppler commented on sechub-pds-solutions/findsecuritybugs/env-initial:
```diff
@@ -7,4 +7,5 @@ JAVA_ENABLE_DEBUG=false
 KEEP_CONTAINER_ALIVE_AFTER_PDS_CRASHED=false
```
This file should be removed. For more information have a look at: https://github.com/mercedes-benz/sechub/pull/1862#issuecomment-1382224019.

Jeeppler reviewed (changes requested):
Thanks for your contribution.

I found a number of issues and described them in: https://github.com/mercedes-benz/sechub/pull/1862#issuecomment-1382224019.

Feel free to ask more questions if you have any.

ddzikon commented:
@Jeeppler thank you for the remarks!

I removed `PDS_VERSION` variable and made scripts use shared `env` files and helper scripts, like it is done in GoSec.  

File `env-cluster-inital` for FindSecurityBugs contained also config variables for database connection.  
Should FindSecurityBugs have separate `env-database` file like there is one in GoSec?  
At the moment, i removed use of this file from scripts `50-start-multiple-docker-compose.sh` and `51-start-multiple-object-storage-docker-compose.sh`.

Please let me know, if there are more adjustments to be done.

Jeeppler commented on sechub-pds-solutions/findsecuritybugs/docker-compose_pds_findsecuritybugs_external-network.yaml:
```diff
@@ -6,13 +6,13 @@ services:
         hostname: pds-findsecuritybugs
         env_file:
             - .env
+            - .emv-single
```
```suggestion
            - .env-single
```

ddzikon replied:
Good catch, sorry for that typo 😅

Jeeppler reviewed (changes requested):
See comment.

Jeeppler commented:
> Should FindSecurityBugs have separate env-database file like there is one in GoSec?

Yes, that file is important. Each Tool+PDS combination needs to have it's own database.

ddzikon commented:
> Yes, that file is important. Each Tool+PDS combination needs to have it's own database.

I recreated this file with configuration based on the contents of old `env-cluster-initial` file and adequate files for GoSec.

Jeeppler reviewed (approved):
Looks good 🥇 . Thanks for your contribution.
